extension.js

```diff
@@ -294,3 +294,135 @@ const getParamsFromGithubRaw = (url) => {
     const installedExtensions = getLocalStorageDataFromKey(LOCALSTORAGE_KEYS.installedExtensions, []);
     installedExtensions.forEach((extensionKey) => initializeExtension(extensionKey));
 })();
+const ITEMS_PER_REQUEST = 100;
+async function storeThemes() {
+    const BLACKLIST = await Blacklist();
+    let url = `https://api.github.com/search/repositories?q=${encodeURIComponent("topic:spicetify-themes")}&per_page=${ITEMS_PER_REQUEST}`;
+    const allRepos = await fetch(url).then(res => res.json()).catch(() => []);
+    if (!allRepos.items) {
+        Spicetify.showNotification("Too Many Requests, Cool Down.");
+    }
+    const filteredResults = {
+        ...allRepos,
+        items: allRepos.items.filter(item => !BLACKLIST.includes(item.html_url)),
+    };
+    return filteredResults;
+}
+async function storeExtensions() {
+    const BLACKLIST = await Blacklist();
+    let url = `https://api.github.com/search/repositories?q=${encodeURIComponent("topic:spicetify-extensions")}&per_page=${ITEMS_PER_REQUEST}`;
+    const allRepos = await fetch(url).then(res => res.json()).catch(() => []);
+    if (!allRepos.items) {
+        Spicetify.showNotification("Too Many Requests, Cool Down.");
+    }
+    const filteredResults = {
+        ...allRepos,
+        items: allRepos.items.filter(item => !BLACKLIST.includes(item.html_url)),
+    };
+    return filteredResults;
+}
+async function Blacklist() {
+    const url = "https://raw.githubusercontent.com/CharlieS1103/spicetify-marketplace/main/blacklist.json";
+    const jsonReturned = await fetch(url).then(res => res.json()).catch(() => { });
+    return jsonReturned.repos;
+}
+(async function initializePreload() {
+
+    // Begin by getting the themes and extensions from github
+    const extensionsArray = await storeExtensions();
+    const themesArray = await storeThemes();
```

Comment:
Maybe run them concurrently. 
```suggestion
    const [extensionsArray, themesArray] = await Promise.all([
        storeExtensions(), 
        storeThemes(),
    ]);
```